internal/test/test.go

```diff
@@ -55,43 +59,60 @@ func run(
 	_ command.GlobalFlags,
 	services *services.Services,
 ) (command.Result, error) {
+	testFiles := make(map[string][]byte, 0)
+	for _, filename := range args {
+		code, err := readerWriter.ReadFile(filename)
 
-	filename := args[0]
+		if err != nil {
+			return nil, fmt.Errorf("error loading script file: %w", err)
+		}
 
-	code, err := readerWriter.ReadFile(filename)
-	if err != nil {
-		return nil, fmt.Errorf("error loading script file: %w", err)
+		testFiles[filename] = code
+	}
+
+	var coverageReport *runtime.CoverageReport
+	if testFlags.Cover {
+		coverageReport = runtime.NewCoverageReport()
 	}
 
 	result, err := services.Tests.Execute(
-		code,
-		filename,
+		testFiles,
 		readerWriter,
+		coverageReport,
 	)
 
 	if err != nil {
 		return nil, err
 	}
 
+	if coverageReport != nil && testFlags.CoverProfile != "" {
```

Comment:
Maybe write to a default file, if the user hasn't provided one.
If not (or maybe always) report a warning if one of the flags is provided but not the other, rather than silently not writing the coverage.